clippy_lints/src/doc/mod.rs

```diff
@@ -1147,7 +1186,8 @@ fn check_doc<'a, Events: Iterator<Item = (pulldown_cmark::Event<'a>, Range<usize
                         // Don't check the text associated with external URLs
                         continue;
                     }
-                    text_to_check.push((text, range, code_level));
+                    text_to_check.push((text, range.clone(), code_level));
+                    doc_suspicious_footnotes::check(cx, doc, range, &fragments);
```

Comment:
I'm sure that you've already tried this, but is there a reason that you didn't use [`FootnoteReference`](https://docs.rs/pulldown-cmark/latest/pulldown_cmark/enum.Event.html#variant.FootnoteReference) from ķ`Event`? You would not have to do span magic and manually parse footnotes and it seems that this function doesn't actually use `text `

Reply:
The `FootnoteReference` event is only emitted if the code *is actually* a footnote. This lint fires for text that looks like a footnote reference *but isn't*.